cogs/annual_handover_and_reset.py

```diff
@@ -0,0 +1,198 @@
+                reason=f"{ctx.user} used TeX-Bot slash-command: \"handover\"",
+            )
+
+            logger.debug("Removing Committee-Elect role from user: %s", committee_elect_member)
+            await committee_elect_member.remove_roles(
+                committee_elect_role,
+                reason=f"{ctx.user} used TeX-Bot slash-command: \"handover\"",
+            )
+
+        await initial_response.edit(content=":white_check_mark: Handover procedure complete!")
+
+
+class AnnualResetRolesCommandCog(TeXBotBaseCog):
+    """Cog class that defines the reset_roles command."""
+
+    YEAR_ROLE_NAMES: Final[frozenset[str]] = frozenset(
+        {
+        "Foundation Year",
+        "First Year",
+        "Second Year",
+        "Final Year",
+        "Year In Industry",
+        "Year Abroad",
+        "PGT",
+        },
+    )
+
+    @discord.slash_command(  # type: ignore[no-untyped-call, misc]
+        name="annual_role_reset",
+        description="Removes member and year roles from all users",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def annual_role_reset(self, ctx: TeXBotApplicationContext) -> None:
+        """
+        Definition & callback response of the "reset_roles" command.
+
+        The "reset_roles" command removes the "Memeber" and respective year roles from
+        any user that has them and subsequently resets the GroupMadeMember database model.
+        """
+        # NOTE: Shortcut accessors are placed at the top of the function, so that the exceptions they raise are displayed before any further errors may be sent
+        main_guild: discord.Guild = self.bot.main_guild
+        member_role: discord.Role = await self.bot.member_role
+
+        logger.debug("Reset roles command called.")
+        initial_response: discord.Interaction | discord.WebhookMessage = await ctx.respond(
+            ":hourglass: Resetting membership and year roles... :hourglass:",
+        )
+
+        member: discord.Member
+        for member in member_role.members:
+            await member.remove_roles(
+                member_role,
+                reason=f"{ctx.user} used TeX-Bot slash-command: \"/reset_roles\"",
+            )
+
+        logger.debug("Removed Member role from all users!")
+        await initial_response.edit(
+            content=":hourglass: Removed member role from all users...",
+        )
+
+        await GroupMadeMember._default_manager.all().adelete()
+
+        await initial_response.edit(
+            content=":white_check_mark: Deleted all members from the database...",
+        )
+        logger.debug("Deleted all members from the database.")
+
+        year_roles: set[discord.Role] = {
+            role
+            for role_name
+            in self.YEAR_ROLE_NAMES
+            if isinstance(
+                (role := discord.utils.get(main_guild.roles, name=role_name)), discord.Role,
```

Comment:
* Expansion of calling function should separate arguments onto an independent line each
* Use less expensive boolean existence check (this ensures the value is truthy and thus is not `None`)

Fix:
```python
...
in self.ACADEMIC_YEAR_ROLE_NAMES
if (role := discord.utils.get(main_guild.roles, name=role_name))
...
```